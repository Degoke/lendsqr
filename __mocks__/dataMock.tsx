import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
