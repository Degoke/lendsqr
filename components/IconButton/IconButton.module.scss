.iconButton {
    border: none;
    background: inherit;
}