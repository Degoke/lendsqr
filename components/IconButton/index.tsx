import Image, { StaticImageData } from 'next/image'
import styles from './iconButton.module.scss'

type Props = {
    image: StaticImageData
    alt: string
}

export default function IconButton({ image, alt }: Props) {
    return (
        <button className={styles.iconButton}>
            <Image src={image} alt={alt} />
        </button>
    )
}