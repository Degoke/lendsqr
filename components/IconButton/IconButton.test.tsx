import { render, screen } from '@testing-library/react'
import IconButton from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('FilterMenu tests', () => {
  it('renders Filtermenu component unchanged', () => {
    const { container } = render(<IconButton image={mockImage} alt={''} />)
    expect(container).toMatchSnapshot()
  })
})
