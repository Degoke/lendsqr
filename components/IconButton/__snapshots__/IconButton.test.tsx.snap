// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu tests renders Filtermenu component unchanged 1`] = `
<div>
  <button
    class="iconButton"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</div>
`;
